docs/project-management/branching-and-versioning.md

```diff
@@ -0,0 +1,35 @@
+# Branching and versioning strategy
+
+## Versioning
+
+kubebuilder-declarative-pattern is a library, that works with controller-runtime and kubernetes.
+
+We follow semantic versioning, similar to other tooling in the kubernetes ecosystem.
+
+Because both controller-runtime and client-go introduce changes that require code changes, we follow their versioning.
+Specifically we align with the controller-runtime version, which itself aligns with kubernetes versioning.
+
+Thus:
+
+| kdp version | controller-runtime version | client-go version |
+|---|---|---|
+| v0.20 | v0.20 | v0.32 |
+| v0.19 | v0.19 | v0.31 |
+| v0.18 | v0.18 | v0.30 |
+| ... | ... | ... | 
+| v0.x | v0.x | v0.x+12 |
```

Comment:
I think we should have the versioning scheme allow for releasing multiple kdp versions per controller-runtime/client-go version; easiest is probably just to increment the patch, but we might want to explicitly call that out here.

Reply:
I'll try to expand.  My 2c is that while in theory we might want to make breaking changes more often than client-go releases, in practice I don't think that's very likely, and I think that would make the library difficult to consume.  So agree that we should have multiple patch releases - agree that's not clear.  But those patches should avoid breaking changes.

Note that we can still make as many breaking changes as we want on the main branch; we just only release a batch of breaking changes every 4 months or so.  (And of course the ideal number of breaking changes is zero :-) )

Reply:
I added a small paragraph here to describe this - thanks for the suggestion!